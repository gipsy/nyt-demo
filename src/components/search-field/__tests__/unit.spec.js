import React from 'react'
import renderer from 'react-test-renderer'
import getElementWithContext from 'react-test-context-provider'
import { store } from '@tests/store'

import SearchField from '@components/search-field'

describe('SearchField', () => {

  it('renders without crashing', () => {
    const el = getElementWithContext({ store }, <SearchField />)
    const component = renderer.create(el).toJSON()
    expect(component).toMatchSnapshot()
  })

  // it('renders actual component', () => {
  //   expect(wrap.find(SearchField).exists()).toBeTruthy()
  // })
  //
  // it('renders button with appropriate text', () => {
  //   expect(wrap.find('button').text()).toEqual('Search')
  // })
})
