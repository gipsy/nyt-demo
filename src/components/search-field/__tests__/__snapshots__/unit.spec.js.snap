// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchField renders without crashing 1`] = `
.c1 {
  display: block;
}

.c0 {
  margin: 0 auto;
  width: 50%;
}

@media (max-width:600px) {
  .c0 {
    width: 100%;
  }
}

<div
  className="c0"
>
  <span
    className="ant-input-search c1 ant-input-search-enter-button ant-input-search-large ant-input-affix-wrapper ant-input-affix-wrapper-lg"
  >
    <input
      className="ant-input ant-input-lg"
      disabled={false}
      onKeyDown={[Function]}
      placeholder="input search text"
      size="large"
      style={null}
      type="text"
    />
    <span
      className="ant-input-suffix"
    >
      <button
        className="ant-btn ant-input-search-button ant-btn-primary ant-btn-lg"
        onClick={[Function]}
        type="button"
      >
        <span>
          Search
        </span>
      </button>
    </span>
  </span>
</div>
`;
